content/vi/docs/concepts/containers/container-lifecycle-hooks.md

```diff
@@ -0,0 +1,123 @@
+---
+reviewers:
+- huynguyennovem
+title: Container Lifecycle Hooks
+content_template: templates/concept
+weight: 30
+---
+
+{{% capture overview %}}
+
+Trang này mô tả cách mà kubelet quản lý các Container có thể sử dụng framework Container lifecycle hook để
+chạy mã nguồn được kích hoạt bởi các sự kiện trong lifecycle của nó.  
+
+{{% /capture %}}
+
+
+{{% capture body %}}
+
+## Tổng quan
+
+Tương tự như nhiều framework ngôn ngữ lập trình có thành phần các lifecycle hooks, như là Angular,
+Kubernetes cung cấp cho Container các lifecycle hook.
+Các hooks cho phép Container nhận thức được các sự kiện trong lifecycle của chúng
+và chạy mã nguồn được triển khai trong một trình xử lý khi các lifecycle hook tương ứng được thực thi.
+
+## Container hooks
+
+Có 2 hooks:
+
+`PostStart`
+
+Hook này thực thi ngay sau khi một container được tạo mới.
+Tuy nhiên, không có gì đảm bảo rằng hook này sẽ thực thi trước container ENTRYPOINT.
+Không có tham số nào được truyền cho trình xử lý (handler).
+
+`PreStop`
+
+Hook này được gọi ngay tức thì ngay trước khi một container bị chấm dứt bởi một API request hoặc quản lý sự kiện như liveness probe failure, preemption, tranh chấp tài nguyên và các vấn đề khác. Một lời gọi tới preStop hook thất bại nếu container đã ở trạng thái kết thúc hoặc đã hoàn thành.
+Nó đang chặn (blocking), có nghĩa là nó đồng bộ,
+vì vậy nó phải hoàn thành trước khi lời gọi xóa container có thể được gửi.
+Không có tham số nào được truyền cho trình xử lý (handler).
+
+Xem thêm chi tiết về hành vi chấm dứt (termination behavior) tại
+[Termination of Pods](/docs/concepts/workloads/pods/pod/#termination-of-pods).
+
+### Các cách thực hiện Hook handler (Hook handler implementations)
+
+Các container có thể truy cập một hook bằng cách thực hiện và đăng ký một handler cho hook đó.
+Có 2 loại hook handler có thể được triển khai cho các containers:
+
+* Exec - Thực thi một lệnh cụ thể, như là `pre-stop.sh` trong cgroups và namespaces của Container.
+Tài nguyên được sử dụng bởi lệnh được tính vào Container.
+* HTTP - Thực thi một HTTP request với một endpoint cụ thể trên Container.
+
+### Thực thi hook handler (Hook handler execution)
+
+Khi một hook quản lý  một Container lifecycle được gọi,
+hệ thống quản lý Kubernetes thực thi trình xử lý (handler) trong Container đã được đăng kí cho hook đó.
+
+Các lời gọi hook handler là đồng bộ trong ngữ cảnh của Pod chứa Container.
+Điều này có nghĩa là đối với hook `PostStart`,
+Container ENTRYPOINT và hook không đồng bộ.
+Tuy nhiên, nếu hook nế hook mất quá nhiều thời gian để chạy hoặc treo,
+Container không thể đạt đến trạng thái `đang chạy`.
```

Comment:
I think should keep `running` instead of `đang chạy`. This will be same as `Terminating` below.